api/src/main/java/org/pmiops/workbench/mailchimp/MailChimpServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package org.pmiops.workbench.mailchimp;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+import com.ecwid.maleorang.MailchimpClient;
+import com.ecwid.maleorang.MailchimpException;
+import com.ecwid.maleorang.method.v3_0.lists.members.EditMemberMethod.Create;
+import com.ecwid.maleorang.method.v3_0.lists.members.GetMemberMethod;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.google.CloudStorageService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+/**
+ * Encapsulate mailchimp API interaction details and provide a simple/mockable interface
+ * for internal use.
+ */
+@Service
+public class MailChimpServiceImpl implements MailChimpService {
+  private static final Logger log = Logger.getLogger(MailChimpServiceImpl.class.getName());
+
+  private final Provider<CloudStorageService> cloudStorageServiceProvider;
+  private String apiKey = null;
+
+  @Autowired
+  public MailChimpServiceImpl(Provider<CloudStorageService> cloudStorageServiceProvider) {
+    this.cloudStorageServiceProvider = cloudStorageServiceProvider;
+  }
+  @Override
+  public String addUserContactEmail(String contactEmail) {
+    String userId;
+    Create createRequest = new Create(
+        cloudStorageServiceProvider.get().readMailChimpListId(),
+        contactEmail);
+    createRequest.status = "pending";
+    try {
+      userId = getClient().execute(createRequest).mapping.get("id").toString();
+    } catch (Exception e) {
```

Comment:
What kinds of exceptions can be thrown here?

Rather than throwing generic RuntimeException, we should probably throw ServerErrorException in the "we don't know how to handle this case" and other exceptions in other cases we expect.

Reply:
I actually left it generic on purpose because I wanted to do a follow up to clean up our Exception handling after security freeze, and this gives me obvious markers. Probably should put a todo in though.